src/hints/syscalls.rs

```diff
@@ -304,3 +306,107 @@ pub fn storage_write(
 
     Ok(())
 }
+
+pub const SET_SYSCALL_PTR: &str = indoc! {r#"
+	ids.os_context = segments.add()
+	ids.syscall_ptr = segments.add()
+
+	syscall_handler.set_syscall_ptr(syscall_ptr=ids.syscall_ptr)"#
+};
+
+pub fn set_syscall_ptr(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let os_context = vm.add_memory_segment();
+    let syscall_ptr = vm.add_memory_segment();
+
+    insert_value_from_var_name(vars::ids::OS_CONTEXT, os_context, vm, ids_data, ap_tracking)?;
+    insert_value_from_var_name(vars::ids::SYSCALL_PTR, syscall_ptr, vm, ids_data, ap_tracking)?;
+
+    let syscall_handler: DeprecatedOsSyscallHandlerWrapper = exec_scopes.get(vars::scopes::SYSCALL_HANDLER)?;
+
+    syscall_handler.set_syscall_ptr(syscall_ptr);
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use blockifier::block_context::{BlockContext, FeeTokenAddresses, GasPrices};
+    use cairo_vm::types::exec_scope::ExecutionScopes;
+    use cairo_vm::types::relocatable::Relocatable;
+    use rstest::{fixture, rstest};
+    use starknet_api::block::{BlockNumber, BlockTimestamp};
+    use starknet_api::core::{ChainId, ContractAddress, PatriciaKey};
+    use starknet_api::hash::StarkHash;
+    use starknet_api::{contract_address, patricia_key};
+
+    use super::*;
+    use crate::ExecutionHelperWrapper;
+
+    #[fixture]
+    fn block_context() -> BlockContext {
+        BlockContext {
+            chain_id: ChainId("SN_GOERLI".to_string()),
+            block_number: BlockNumber(1_000_000),
+            block_timestamp: BlockTimestamp(1_704_067_200),
+            sequencer_address: contract_address!("0x0"),
+            fee_token_addresses: FeeTokenAddresses {
+                eth_fee_token_address: contract_address!("0x1"),
+                strk_fee_token_address: contract_address!("0x2"),
+            },
+            vm_resource_fee_cost: Arc::new(HashMap::new()),
+            gas_prices: GasPrices { eth_l1_gas_price: 1, strk_l1_gas_price: 1 },
+            invoke_tx_max_n_steps: 1,
+            validate_max_n_steps: 1,
+            max_recursion_depth: 50,
+        }
+    }
+
+    #[fixture]
+    fn exec_scopes(block_context: BlockContext) -> ExecutionScopes {
```

Comment:
Is this meant to inject `exec_scopes` that should always exist? It should probably be shared outside of `syscalls.rs` in that case. A short comment would help as well.

Reply:
This is (for the moment) specific to this syscall, it will probably be used in more places but I'm not sure yet.